### PR TITLE
/sura and /download

### PR DESCRIPTION
**Pull request by sabeurthabti**
screen shot 2016-11-15 at 09 24 14

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://dev.quranicaudio.com:32887

File: src/reducers/audioplayer.js [hunk 1]
```diff
@@ -106,27 +114,60 @@ export default function reducer(state = initialState, action = {}) {
       };
     case NEXT: {
       state.file.pause();
-      const file = new Audio(`${AUDIO_URL}/${state.qari.relativePath}${zeroPad(state.surah.id + 1, 3)}.mp3`);
+      let file = null;
+      let newState = {};
+      if (state.surahPage) {
+        const nextQari = state.qaris[state.qari.id + 1];
```
Comment (sabeurthabti) [inline src/reducers/audioplayer.js]: not proud of this.

**Comment by ahmedre** [comment]
Deployed to: http://dev.quranicaudio.com:32888

**Comment by ahmedre** [comment]
Deployed to: http://dev.quranicaudio.com:32889

**Comment by ahmedre** [comment]
Deployed to: http://dev.quranicaudio.com:32890

**Comment by ahmedre** [comment]
Deployed to: http://dev.quranicaudio.com:32891

**Comment by ahmedre** [comment]
Deployed to: http://dev.quranicaudio.com:32892

**Comment by ahmedre** [comment]
Deployed to: http://dev.quranicaudio.com:32893

**Comment by ahmedre** [comment]
Deployed to: http://dev.quranicaudio.com:32894

File: src/containers/Qari/Sura.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    surahs: PropTypes.object.isRequired,
+    surah: PropTypes.object.isRequired,
+    currentQari: PropTypes.object.isRequired,
+    state: PropTypes.object.isRequired,
+    files: PropTypes.object.isRequired,
+    load: PropTypes.func.isRequired,
+    play: PropTypes.func.isRequired,
+    params: PropTypes.any,
+    next: PropTypes.func.isRequired,
+    Playing: PropTypes.bool.isRequired,
```
Comment (mmahalwy) [inline src/containers/Qari/Sura.js]: `Playing` capitalized?

File: src/containers/Qari/Sura.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  };
+
+  handleSurahSelection = (qari) => {
+    const { surah, qaris} = this.props;
+
+    this.props.load({ qari, surah, surahPage: true, qaris});
+  }
+
+  render() {
+    const { surah, Playing, qaris, currentQari} = this.props;
```
Comment (mmahalwy) [inline src/containers/Qari/Sura.js]: space after `currentQari`

File: src/reducers/audioplayer.js (hunk 1 above)
Comment (mmahalwy) [inline src/reducers/audioplayer.js]: all is good

**Pull request merged**